### PR TITLE
Adding unit test to parse every JSON in externalized dir

### PR DESCRIPTION
**Pull request by ja2-derek**
To prevent JSON syntax errors. Some files are not always processed in loadGameData(), in particular the localization files.

No other JSON problems found other than those already addressed in [#1115](https://github.com/ja2-stracciatella/ja2-stracciatella/pull/1115/files?w=1).

### CODE REVIEW

**Comment by ja2-derek** [comment]
Linux and mingw64 build fine, just need the JSON fix for tests to pass.

Mac build is failing due to missing `stdc++fs` library. I will check if this can be easily fixed, or else may re-implement the FS stuff with primitives, avoiding `<filesystem>` totaly.

File: src/externalized/DefaultContentManager_unittests.cc [hunk 1]
```diff
@@ -127,4 +137,37 @@ TEST(TempFiles, deleteFile)
...
+TEST(ExternalizedData, readAllData)
+{
+	DefaultContentManager* cm = DefaultContentManagerUT::createDefaultCMForTesting();
+	ASSERT_TRUE(cm->loadGameData());
+}
+
+std::vector<ST::string> recursivelyListAllFiles(ST::string dir, std::string suffix)
+{
+	std::vector<ST::string> files;
+	for (auto& p : fs::recursive_directory_iterator(dir.to_std_string()))
```
Comment (lynxlynxlynx) [inline src/externalized/DefaultContentManager_unittests.cc:149]: We have nothing in FileMan or our rust fs layer to do this? The whole std::filesystem include block looks fragile. CC @flaviojs

Comment (ja2-derek) [reply src/externalized/DefaultContentManager_unittests.cc]: How about now? Added `ls -R` to FileMan

Comment (lynxlynxlynx) [reply src/externalized/DefaultContentManager_unittests.cc]: Looks better to me, thanks. If this is now done and Flávio doesn't comment in a while, go ahead and merge it.

**Comment by ja2-derek** [comment]
Rebased to master

**Comment by lynxlynxlynx** [comment]
I suspect you missed the last commits, since the test breaks on the same json.

**Pull request merged**